apps/betterangels-backend/clients/schema.py

```diff
@@ -32,31 +32,74 @@
     UpdateClientProfileInput,
 )
 
-CLIENT_RELATED_CLS_NAME_BY_RELATED_NAME = {
-    "contacts": "ClientContact",
-    "hmis_profiles": "HmisProfile",
-    "household_members": "ClientHouseholdMember",
-    "social_media_profiles": "SocialMediaProfile",
+CLIENT_RELATED_CLS_NAME_BY_RELATED_NAME: dict[str, dict[str, Any]] = {
+    "contacts": {
+        "module": "clients",
+        "cls_name": "ClientContact",
+        "generic_relation": False,
+    },
+    "hmis_profiles": {
+        "module": "clients",
+        "cls_name": "HmisProfile",
+        "generic_relation": False,
+    },
+    "household_members": {
+        "module": "clients",
+        "cls_name": "ClientHouseholdMember",
+        "generic_relation": False,
+    },
+    "phone_numbers": {
+        "module": "common",
+        "cls_name": "PhoneNumber",
+        "generic_relation": True,
+    },
+    "social_media_profiles": {
+        "module": "clients",
+        "cls_name": "SocialMediaProfile",
+        "generic_relation": False,
+    },
 }
 
 
 def upsert_or_delete_client_related_object(
     info: Info,
+    module: str,
     model_cls_name: str,
+    generic_relation: bool,
     data: List[Dict[str, Any]],
     client_profile: ClientProfile,
 ) -> None:
     """Creates, updates, or deletes a client's related objects.
 
     Expects a list of related objects. Missing elements will be deleted.
     """
-    model_cls = apps.get_model("clients", model_cls_name)
+    model_cls = apps.get_model(module, model_cls_name)
 
     item_updates_by_id = {item["id"]: item for item in data if item.get("id")}
     items_to_create = [item for item in data if not item.get("id")]
-    items_to_update = model_cls.objects.filter(id__in=item_updates_by_id.keys(), client_profile=client_profile)
+    items_to_update = model_cls.objects.filter(id__in=item_updates_by_id.keys())
     model_cls.objects.exclude(id__in=item_updates_by_id).delete()
 
+    if generic_relation:
```

Comment:
not sure if this is overloading the method and we should split this up differently

Reply:
This function is getting pretty big.